lib/index.js

```diff
@@ -126,6 +155,7 @@ Ramen.prototype.identify = function(identify){
 
   // For tracking purposes
   ramenSettings._partner = 'segment.com';
+  ramenSettings.disable_location_watch = true;
```

Comment:
> Ramen doesn't care about anything until #identify is called.

Interesting. Totally valid especially given your use case. In the future you might consider an anonymousId approach, where you later link identified users to an anonymousId—though that totally depends on how you want Ramen to work :stuck_out_tongue: 

> So I think what you're saying is that I should put checks in place on Ramen.prototype.(page|track|group) that make them a no-op until #identify has been called. That right?

Yeah, basically just exit early when a precondition isn't met (in this case, a userId). Something like this:

```js
Ramen.prototype.page = function(page) {
  // here we need to check the analytics.user object, page calls don't carry user data
  if (!this.analytics.user().id()) {
    this.debug('#page failed: no userId provided');
    return;
  }
  // ...
};

Ramen.prototype.group = function(group) {
  if (!group.userId()) {
    this.debug('#group failed: no userId provided');
    return;
  }
  // ...
};
```

Sounds like we'll want to do that in `#identify`, `#group`, and `#page`. We should also update the Ramen documentation on https://segment.com/docs/integrations/ramen/ so users know they must call `#identify` before any other calls. (@amillet89 can help with this.)

> As I re-read this, it hits me: are Segment users instructed to call #group only after #identify

Not _exactly_; `#group` is intended to link a user to a group, so the expectation is that you're calling `#group` on a user that already exists. It's the user's responsibility, though, to ensure that they're `group`ing a user that already exists. There's no need for you to correct for this by e.g. queueing the request and sending it later; that's the user's responsibility.